verticapy/learn/memmodel.py

```diff
@@ -1527,44 +1528,254 @@ def sql_from_clusters(
+            for idx in range(len(X)):
+                val, centroid_val = X[idx], centroid[idx]
+                try:
+                    val = float(val)
+                    centroid_val = float(centroid_val)
+                except:
+                    pass
+                if isinstance(centroid_val, str) or centroid_val == None:
+                    distance_cat += abs(int(val == centroid_val) - 1)
+                else:
+                    distance_num += (val - centroid_val) ** p
+            distance_final = distance_num + gamma * distance_cat
+            result += [distance_final]
+        return result
+
+    result = np.apply_along_axis(compute_distance_row, 1, X)
+
+    if return_proba:
+        result = 1 / (result + 1e-99) / np.sum(1 / (result + 1e-99), axis=1)[:, None]
+    elif not (return_distance_clusters):
+        result = np.argmin(result, axis=1)
+
+    return result
+
+
+# ---#
+def sql_from_clusters_kprotypes(
+    X: list,
+    clusters: list,
+    return_distance_clusters: bool = False,
+    return_proba: bool = False,
+    p: int = 2,
+    gamma: float = 1.0,
+    is_categorical: list = [],
+) -> Union[list, str]:
+    """
+    ---------------------------------------------------------------------------
+    Returns the SQL code needed to deploy a k-means or nearest centroids model 
+    using its attributes.
+
+    Parameters
+    ----------
+    X: list
+        The names or values of the input predictors.
+    clusters: list
+        List of the model's cluster centers.
+    return_distance_clusters: bool, optional
+        If set to True, the distance to the clusters is returned.
+    return_proba: bool, optional
+        If set to True, the probability to belong to the clusters is returned.
+    p: int, optional
+        The p corresponding to the one of the p-distances.
+    gamma: float, optional
+        Weighing factor for categorical columns. It can determine relative 
+        importance of numerical and categorical attributes.
+    is_categorical: list, optional
+        List of booleans to indicates if X[idx] is a categorical variable
+        or not. If empty all the variables are considered as categorical.
+
+    Returns
+    -------
+    str / list
+        SQL code
+    """
+    check_types(
+        [
+            ("X", X, [list, np.ndarray]),
+            ("clusters", clusters, [list, np.ndarray]),
+            ("return_distance_clusters", return_distance_clusters, [bool]),
+            ("return_proba", return_proba, [bool]),
+            ("p", p, [int]),
+            ("gamma", gamma, [float]),
+            ("is_categorical", is_categorical, [list, np.ndarray]),
+        ]
+    )
+
+    assert not (return_distance_clusters) or not (return_proba), ParameterError(
+        "Parameters 'return_distance_clusters' and 'return_proba' cannot both be set to True."
+    )
+
+    if not (is_categorical):
+        is_categorical = [True for i in range(len(X))]
+
+    for c in clusters:
```

Comment:
Shouldn't we have the same check in the predict_from_clusters_kprotypes function?